homeassistant/components/control4/__init__.py

```diff
@@ -49,83 +46,52 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Control4 from a config entry."""
     hass.data.setdefault(DOMAIN, {})
     entry_data = hass.data[DOMAIN].setdefault(entry.entry_id, {})
```

Comment:
The amount of key/values is getting rather large here.  It would be good to use a `dataclass` at this point.

Example
https://github.com/home-assistant/core/blob/dev/homeassistant/components/lookin/__init__.py#L155
https://github.com/home-assistant/core/blob/dev/homeassistant/components/lookin/models.py#L12